### PR TITLE
Sequence number shoult not be set if snapshot is unhandled #23469

### PR DESCRIPTION
**Pull request by nvollmar**
Adds check to correctly recover if snapshot is unhandled

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #23469

**Comment by patriknw** [comment]
OK TO TEST

**Review by patriknw — APPROVED** [review]
LGTM, I see what you mean

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala [hunk 1]
```diff
@@ -462,9 +462,14 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
...
+            if (recoveryBehavior.isDefinedAt(offer)) {
+              setLastSequenceNr(metadata.sequenceNr)
+              // Since we are recovering we can ignore the receive behavior from the stack
+              Eventsourced.super.aroundReceive(recoveryBehavior, offer)
+            } else {
+              unhandled(offer)
+            }
         }
         changeState(recovering(recoveryBehavior, timeout))
         journal ! ReplayMessages(lastSequenceNr + 1L, toSnr, replayMax, persistenceId, self)
```
Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: here it will replay all events if the snapshot is ignored 👍

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
In another PR I noticed that CLA validator says that nvollmar has not signed the CLA, but he is in the list (3 times).

**Comment by nvollmar** [comment]
Strange, because the CLA validator in this pull request green-lighted it

**Comment by patriknw** [comment]
yes, must be a glitch or bug in the validator

**Pull request merged**